src/signalstats.jl

```diff
@@ -50,7 +50,7 @@ function _signalstats_impl(X::AbstractArray{<:RealQuantity}, Y::AbstractArray{<:
     mean_X = sum_X * inv_n
     mean_Y = sum_Y * inv_n
     var_X = sum_X_sqr * inv_n - mean_X * mean_X
-    var_Y = sum_Y_sqr * inv_n - mean_Y * mean_Y
+    var_Y = max(sum_Y_sqr * inv_n - mean_Y * mean_Y, zero(mean_Y))
     cov_XY = sum_XY * inv_n - mean_X * mean_Y
```

Comment:
Just out of curiosity: why are `mean_X`, `var_X` and `cov_XY`calculated if they are not returned?

Reply:
I have no idea. Code is originally from @oschulz

Reply:
You can comment it out, I think it was in case we want `cov_XY` at some point.

Reply:
Done. Is it ok from your side then?